pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxReceiveOperator.java

```diff
@@ -88,8 +88,8 @@ public MailboxReceiveOperator(MailboxService<TransferableBlock> mailboxService,
 
     _exchangeType = exchangeType;
     if (_exchangeType == RelDistribution.Type.SINGLETON) {
-      ServerInstance singletonInstance = null;
-      for (ServerInstance serverInstance : sendingStageInstances) {
+      VirtualServer singletonInstance = null;
+      for (VirtualServer serverInstance : sendingStageInstances) {
```

Comment:
looks like this will be addressed follow up when there's more than 1 virtual server associate with the same physical server.

Reply:
yes that's correct as well